### PR TITLE
refactor: Renames MONGODB_ATLAS_ENABLE_BETA to MONGODB_ATLAS_ENABLE_PREVIEW for features in preview

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Adds MONGODB_ATLAS_ENABLE_PREVIEW flag for features in preview

Link to any related issue(s): CLOUDP-223391

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .github/workflows/migration-tests.yml [hunk 1]
```diff
@@ -265,6 +265,7 @@ jobs:
           terraform_wrapper: false    
       - name: Migration Tests
         env:
+          MONGODB_ATLAS_ENABLE_PREVIEW: ${{ vars.MONGODB_ATLAS_ENABLE_PREVIEW }}
           MONGODB_ATLAS_ENABLE_BETA: ${{ vars.MONGODB_ATLAS_ENABLE_BETA }}
```
Comment (maastha) [inline .github/workflows/migration-tests.yml]: [q] Why are we retaining `MONGODB_ATLAS_ENABLE_BETA ` here but not in the other workflows?

Comment (EspenAlbert) [reply .github/workflows/migration-tests.yml]: Here we are using an earlier version of the atlas terraform provider to check the migration tests. That earlier version will [fail](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/8231528028/job/22507253291)  if the variable is missing. 
For the other acceptance tests they all use the current version of the provider; therefore, they can use the new env-variable name

Comment (lantoli) [reply .github/workflows/migration-tests.yml]: that's ok

another alternative is to use SkipIfVersionBelow and/or IsProviderVersionAtLeast so we don't run those mig tests in prev. version to this one.

File: internal/provider/provider.go [hunk 2]
```diff
@@ -414,14 +414,14 @@ func (p *MongodbtlasProvider) DataSources(context.Context) []func() datasource.D
...
-	betaDataSources := []func() datasource.DataSource{
+	previewDataSources := []func() datasource.DataSource{
 		streaminstance.DataSource,
 		streaminstance.PluralDataSource,
 		streamconnection.DataSource,
 		streamconnection.PluralDataSource,
 	}
-	if ProviderEnableBeta {
-		dataSources = append(dataSources, betaDataSources...)
+	if ProviderEnablePreview || ProviderEnableBeta {
```
Comment (lantoli) [inline internal/provider/provider.go]: are we keeping ProviderEnableBeta in the condition ?

File: internal/provider/provider_sdk2.go [hunk 3]
```diff
@@ -62,7 +62,8 @@ import (
 )
 
 var (
-	ProviderEnableBeta, _ = strconv.ParseBool(os.Getenv("MONGODB_ATLAS_ENABLE_BETA"))
+	ProviderEnableBeta, _    = strconv.ParseBool(os.Getenv("MONGODB_ATLAS_ENABLE_BETA"))
```
Comment (lantoli) [inline internal/provider/provider_sdk2.go]: are we keeping beta vars?

Comment (EspenAlbert) [reply internal/provider/provider_sdk2.go]: Yes, but hopefully not for long. Based on the comments in the Jira Issue:
>  when Streams has GA [...] we can then remove MONGODB_ATLAS_ENABLE_BETA flag completely

File: README.md [hunk 4]
```diff
@@ -27,12 +27,12 @@ To use a released provider in your Terraform environment, run [`terraform init`]
...
 
-## Beta Features
-In order to use and/or test beta resources and datasources in this provider you'll need to set the environment variable `MONGODB_ATLAS_ENABLE_BETA` to true.
+## Preview Features
+In order to use and/or test preview resources and datasources in this provider you'll need to set the environment variable `MONGODB_ATLAS_ENABLE_PREVIEW` to true.
 
 
 ```bash
-export MONGODB_ATLAS_ENABLE_BETA = true
+export MONGODB_ATLAS_ENABLE_PREVIEW = true
```
Comment (Zuhairahmed) [inline README.md]:
```suggestion
export MONGODB_ATLAS_ENABLE_PREVIEW=true
```

believe there shouldn't be spaces before and after the '=' sign in unix based terminals

File: .github/workflows/migration-tests.yml (hunk 1 above)
Comment (maastha) [reply .github/workflows/migration-tests.yml]: Whichever approach we take, could we add a comment to capture this?

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: .github/workflows/migration-tests.yml (hunk 1 above)
Comment (EspenAlbert) [reply .github/workflows/migration-tests.yml]: Added a comment in a2bc4ed

File: internal/provider/provider.go (hunk 2 above)
Comment (EspenAlbert) [reply internal/provider/provider.go]: Not kept anymore

File: internal/provider/provider_sdk2.go (hunk 3 above)
Comment (EspenAlbert) [reply internal/provider/provider_sdk2.go]: Not anymore :D

File: .github/workflows/migration-tests.yml (hunk 1 above)
Comment (EspenAlbert) [reply .github/workflows/migration-tests.yml]: @maastha ok to resolve?

File: .github/workflows/migration-tests.yml [hunk 5]
```diff
@@ -265,6 +265,8 @@ jobs:
           terraform_wrapper: false    
       - name: Migration Tests
         env:
+          MONGODB_ATLAS_ENABLE_PREVIEW: ${{ vars.MONGODB_ATLAS_ENABLE_PREVIEW }}
+          # can be removed when MONGODB_ATLAS_LAST_VERSION supports the new MONGODB_ATLAS_ENABLE_PREVIEW flag
           MONGODB_ATLAS_ENABLE_BETA: ${{ vars.MONGODB_ATLAS_ENABLE_BETA }}
```
Comment (lantoli) [inline .github/workflows/migration-tests.yml]: can we change to MONGODB_ATLAS_ENABLE_BETA: ${{ vars. MONGODB_ATLAS_ENABLE_PREVIEW }}

so we only have the GH var for preview?

**Review by lantoli — APPROVED** [review]

File: .github/workflows/migration-tests.yml (hunk 5 above)
Comment (lantoli) [reply .github/workflows/migration-tests.yml]: in fact not sure why do we have those vars, can we just do:
  MONGODB_ATLAS_ENABLE_PREVIEW: true
 MONGODB_ATLAS_ENABLE_BETA: true

and similarly in the other places?

i don't see value in having those cc @AgustinBettati

Comment (EspenAlbert) [reply .github/workflows/migration-tests.yml]: done in 7597c2c ✅ 
*Update: saw the last comment, and added a86ed15

**Review by Zuhairahmed — APPROVED** [review]

**Review by maastha — APPROVED** [review]
LGTM

File: .github/workflows/migration-tests.yml [hunk 6]
```diff
@@ -265,7 +265,9 @@ jobs:
           terraform_wrapper: false    
       - name: Migration Tests
         env:
-          MONGODB_ATLAS_ENABLE_BETA: ${{ vars.MONGODB_ATLAS_ENABLE_BETA }}
+          MONGODB_ATLAS_ENABLE_PREVIEW: "true"
```
Comment (lantoli) [inline .github/workflows/migration-tests.yml:268]: knitpit: I think "true" and true  are equivalent here as env vars are passed as strings

File: .github/workflows/acceptance-tests-runner.yml [hunk 7]
```diff
@@ -306,7 +306,7 @@ jobs:
           terraform_wrapper: false  
       - name: Acceptance Tests
         env:
-          MONGODB_ATLAS_ENABLE_BETA: ${{ vars.MONGODB_ATLAS_ENABLE_BETA }}
+          MONGODB_ATLAS_ENABLE_PREVIEW: ${{ vars.MONGODB_ATLAS_ENABLE_PREVIEW }}
```
Comment (lantoli) [inline .github/workflows/acceptance-tests-runner.yml:309]: vars.MONGODB_ATLAS_ENABLE_PREVIEW can be replaced by true as in mig config

we need to remove all references to vars.MONGODB_ATLAS_ENABLE_PREVIEW and vars.MONGODB_ATLAS_ENABLE_BETA so they can be removed

File: .github/workflows/examples.yml [hunk 8]
```diff
@@ -13,7 +13,7 @@ on:
 
 env:
   AWS_DEFAULT_REGION: us-west-2
-  MONGODB_ATLAS_ENABLE_BETA: ${{ vars.MONGODB_ATLAS_ENABLE_BETA }}
+  MONGODB_ATLAS_ENABLE_PREVIEW: ${{ vars.MONGODB_ATLAS_ENABLE_PREVIEW }}
```
Comment (lantoli) [inline .github/workflows/examples.yml:16]: same, vars.MONGODB_ATLAS_ENABLE_PREVIEW can be replaced by true

File: .github/workflows/acceptance-tests-runner.yml (hunk 7 above)
Comment (EspenAlbert) [reply .github/workflows/acceptance-tests-runner.yml:309]: Sorry, I didn't see these changes before merging 🤦
Maybe we can clean it up in one of the test  improvement PRs?

**Pull request merged**